cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -122,6 +122,11 @@ func (pe *ProbeEvent) deleteBlockDeviceEvent(msg controller.EventMessage) {
 		if isGPTBasedUUIDEnabled {
 			_ = pe.deleteBlockDevice(*device, bdAPIList)
 		} else {
+			if device.DeviceAttributes.DeviceType == libudevwrapper.UDEV_PARTITION {
```

Comment:
@liuminjian Is this added so that we can avoid a search in the BDList? 

Since resources for partitions wont be created if GPTBasedUUID is disabled, the `GetExistingBlockDeviceResource()` will return nil.

So why is this check required?

Reply:
I want to provide the entire disk instead of the disk partition to the container. I don't want to create block device resources for partitions. However, if you format and partition the disk in the container, a partition deletion event may occur, the block device will be set inactived.


![image](https://user-images.githubusercontent.com/7840869/132125589-b63fc2ee-4335-42ad-a595-624fb4f1291b.png)

Reply:
Can you replace this with [partition](https://github.com/openebs/node-disk-manager/blob/develop/blockdevice/blockdevice.go#L210) from here. We want to use the blockdevice API and restrict the usage of udev package to udev related tasks only.

Reply:
@liuminjian can you take a look at the above comment.

Reply:
I replaced with blockdevice.BlockDeviceTypePartition